components/cluster-coordinator/org.wso2.carbon.cluster.coordinator/org.wso2.carbon.cluster.coordinator.rdbms/src/main/java/org/wso2/carbon/cluster/coordinator/rdbms/RDBMSCoordinationStrategy.java

```diff
@@ -35,79 +38,134 @@
+                    get(CoordinationPropertyNames.STRATEGY_CONFIG_NS);
+            if (strategyConfiguration != null) {
+                this.heartBeatInterval = (int) strategyConfiguration.
+                        getOrDefault(RDBMSConstants.HEART_BEAT_INTERVAL, 1000);
+                // Maximum age of a heartbeat. After this much of time, the heartbeat is considered invalid and node is
+                // considered to have left the cluster.
+                this.heartbeatMaxAge = heartBeatInterval *
+                        (int) strategyConfiguration.getOrDefault(RDBMSConstants.HEART_BEAT_MAX_AGE, 2);
+                this.localGroupId = (String) clusterConfiguration.get(CoordinationPropertyNames.GROUP_ID_PROPERTY);
+            } else {
+                throw new ClusterCoordinationException("Strategy Configurations not found in" +
+                        " deployment.yaml, please check " + CoordinationPropertyNames.STRATEGY_CONFIG_NS + " under "
+                        + CoordinationPropertyNames.CLUSTER_CONFIG_NS + " namespace configurations");
+            }
+        } else {
+            throw new ClusterCoordinationException("Cluster Configurations not found in" +
+                    " deployment.yaml, please check " + CoordinationPropertyNames.CLUSTER_CONFIG_NS +
+                    " namespace configurations");
+        }
+        if (this.localGroupId == null) {
+            throw new ClusterCoordinationException(
+                    "Group Id not set in cluster.config in deployment.yaml configuration file");
+        }
         ThreadFactory namedThreadFactory = new ThreadFactoryBuilder()
                 .setNameFormat("RDBMSCoordinationStrategy-%d").build();
-        this.threadExecutor = Executors.newScheduledThreadPool(
-                CoordinationStrategyConfiguration.getInstance().getRdbmsConfigs()
-                        .get("heartbeatInterval"), namedThreadFactory);
-        this.heartBeatInterval = CoordinationStrategyConfiguration.getInstance().getRdbmsConfigs()
-                .get(CoordinationPropertyNames.RDBMS_BASED_COORDINATION_HEARTBEAT_INTERVAL);
-        // Maximum age of a heartbeat. After this much of time, the heartbeat is considered invalid and node is
-        // considered to have left the cluster.
-        this.heartbeatMaxAge = heartBeatInterval * 2;
-        this.localNodeId = generateRandomId();
-        this.rdbmsMemberEventProcessor = new RDBMSMemberEventProcessor(localNodeId, datasource);
-        this.communicationBusContext = new RDBMSCommunicationBusContextImpl(datasource);
+        this.threadExecutor = Executors.newSingleThreadScheduledExecutor(namedThreadFactory);
+        try {
+            ConfigProvider configProvider = RDBMSCoordinationServiceHolder.getConfigProvider();
+            if (configProvider != null) {
+                this.localNodeId = (String) configProvider.getConfigurationMap("wso2.carbon").get("id");
```

Comment:
wso2.carbon - check whether this constant is defined in the kernel. If so use that constant instead of hard-coding

Reply:
There is no constant used. Namespace is hard coded in org.wso2.carbon.kernel.config.model.CarbonConfiguration

Reply:
Ack